web_timeline/static/src/js/web_timeline.js

```diff
@@ -0,0 +1,514 @@
+            return this._super();
+        },
+
+        is_action_enabled: function(action) {
+            if (action === 'create' && !this.options.creatable) {
+                return false;
+            }
+            return this._super(action);
+        },
+
+        on_add: function(item, callback) {
+            var self = this;
+            var context = this.dataset.get_context();
+            // Initialize default values for creation
+            var default_context = {}
+            default_context['default_'.concat(this.date_start)] = item.start;
+            default_context['default_'.concat(this.date_stop)] = moment(item.start).add(1, 'hours').toDate();
+            if (item.group > 0) {
+                default_context['default_'.concat(this.last_group_bys[0])] = item.group;
+            }
+            context.add(default_context);
+            // Show popup
+            var dialog = new form_common.FormViewDialog(this, {
+                res_model: this.dataset.model,
+                res_id: null,
+                context: context,
+                view_id: +this.open_popup_action,
+            }).open();
+            dialog.on('create_completed', self, function(id) {
+                self.dataset.ids = this.dataset.ids.concat([id]);
+                self.dataset.trigger("dataset_changed", id);
+                self.dataset.read_ids([id], self.fields).done(function(records) {
+                    var new_event = self.event_data_transform(records[0]);
+                    var items = self.timeline.itemsData;
+                    items.add(new_event);
+                    self.timeline.setItems(items);
+                });
+            });
+            return false;
+        },
+
+        on_update: function(item, callback) {
+            var self = this;
+            var id = item.evt.id;
+            var title = item.evt.__name;
+            if (! this.open_popup_action) {
+                var index = this.dataset.get_id_index(id);
+                this.dataset.index = index;
+                if (this.write_right) {
+                    this.do_switch_view('form', null, { mode: "edit" });
+                } else {
+                    this.do_switch_view('form', null, { mode: "view" });
+                }
+            }
+            else {
+                var dialog = new form_common.FormViewDialog(this, {
+                    res_model: this.dataset.model,
+                    res_id: parseInt(id).toString() == id ? parseInt(id) : id,
+                    context: this.dataset.get_context(),
+                    title: title,
+                    view_id: +this.open_popup_action,
+                }).open();
+                dialog.on('write_completed', self, function() {
+                    self.dataset.trigger("dataset_changed", id);
+                    self.current_window = self.timeline.getWindow();
+                    self.reload();
+                    self.timeline.setWindow(self.current_window);
+                });
```

Comment:
Again, you can define a `on_write_completed` method in the object and call it here with a proxy, to let it be inherited/extended.